### ISSUE TITLE
RTL support for Hebrew, Arabic, Persian, Urdu

### ISSUE THREAD

**Issue by Netanel-M** [movim/movim#873]
Hebrew, Arabic, Persian and Urdu are written from the right to the left. Movim already seems to have partial support for RTL. Chat messages written in Hebrew are received and sent correctly, however when writing them they are aligned to the left instead of to the right. Also special symbols at the end of the line appear at the opposite side.
`.it looks like this`
When a message is sent however, the message seems to be encoded correctly causing the symbol to appear in the right side.
When it come to posts however, the problem is trickier. Not only the text is aligned to the left and symbols appear at the wrong side of the line when writing a post, but persist when the post is published.
To solve this problem in php is quite easy, [simply call the hebrev function with the text](https://www.php.net/manual/en/function.hebrev.php)
This combined with aligning the text to the right fully implements hebrew support.
More research is required to find out whether Persian and Arabic have the same issues or different ones.

I realize Movim mostly has a European community making this request low priority, but this is a barrier preventing from millions of people from potentially adopting it as a social media platform.

**Comment by edhelas** [comment]
I fully agree and would be pleased to check support or RTL content in Movim :)
Would it be possible to have some nice examples to work with (articles and chat) ? You can contact me also directly then I can chat with you and get those content this way.

**Comment by Netanel-M** [comment]
> I fully agree and would be pleased to check support or RTL content in Movim :)
> Would it be possible to have some nice examples to work with (articles and chat) ? You can contact me also directly then I can chat with you and get those content this way.

Great, how can I contact you directly?

**Comment by edhelas** [comment]
Using my XMPP account edhelas _ at _ movim _ dot _ eu

**Comment by edhelas** [comment]
Any news on that one ?

**Comment by Netanel-M** [comment]
The easiest way to implement this would be having the browser handle it for example:
```<element dir="auto"> ```
While the text is still encoded incorrectly, it wouldn't matter since it would be amended client-side by the browser.
[There's excellent documentation for the direction attribute on the world wide web consortium website](https://www.w3.org/International/questions/qa-html-dir)
[Here is the mozilla documentation for the dir attribute with a quick snippet](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir)

I wanted to work on it myself, but couldn't get a development environment working.

**Comment by edhelas** [comment]
I started some work in the UI. It's more complicated than in the links because I have to do custom CSS.
What you could do to help me is complete some translation in right to left languages on our translation platform https://www.transifex.com/movim/movim/content/ .

**Comment by edhelas** [comment]
Closing the ticket as more work has been put to support RTL, its "good enough" at the moment.
